OpenRA.Mods.Common/Traits/CombatDebugOverlay.cs

```diff
@@ -50,48 +52,42 @@ void INotifyCreated.Created(Actor self)
 			allBlockers = self.TraitsImplementing<IBlocksProjectiles>().ToArray();
 		}
 
-		void IRenderAboveWorld.RenderAboveWorld(Actor self, WorldRenderer wr)
+		IEnumerable<IRenderable> IRenderAboveShroud.RenderAboveShroud(Actor self, WorldRenderer wr)
 		{
-			if (debugVis == null || !debugVis.CombatGeometry)
-				return;
+			if (debugVis == null || !debugVis.CombatGeometry || self.World.FogObscures(self))
+				yield break;
 
 			var wcr = Game.Renderer.WorldRgbaColorRenderer;
 			var iz = 1 / wr.Viewport.Zoom;
 
 			var blockers = allBlockers.Where(Exts.IsTraitEnabled).ToList();
 			if (blockers.Count > 0)
 			{
-				var hc = Color.Orange;
 				var height = new WVec(0, 0, blockers.Max(b => b.BlockingHeight.Length));
-				var ha = wr.Screen3DPosition(self.CenterPosition);
-				var hb = wr.Screen3DPosition(self.CenterPosition + height);
-				wcr.DrawLine(ha, hb, iz, hc);
-				TargetLineRenderable.DrawTargetMarker(wr, hc, ha);
-				TargetLineRenderable.DrawTargetMarker(wr, hc, hb);
```

Comment:
Is the idea to remove the TargetMarker from here?

Reply:
Yeah, I didn't think this gave enough value to justify creating another renderable for it.